# Project 3 -- [Drew] [Caldwell]

**TA Help:** N/A

**Collaboration:** N/A


## Question 1

Here we are laoding a data set using the fread

In [6]:
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/flights/subset/1990.csv")

## Question 2

Here we are loading the dataset then using tapply to find the sum for the 4 different regions of stores transactions

In [2]:
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/8451/The_Complete_Journey_2_Master/5000_transactions.csv")

In [3]:
tapply(myDF$SPEND, myDF$STORE_R, sum, na.rm = TRUE)

CENTRAL     EAST    SOUTH     WEST 
 8897305 11699447  7957921  9680107

We see here that the East region spent the most, and the South region spent the least.

## Question 3

Here we are going to load the dataset as before, transform the date so that we have a year variable that we can sort by, and then sum the years 2016 and 2017

In [1]:
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/8451/The_Complete_Journey_2_Master/5000_transactions.csv")

Now I am going to create a new table that will have the total spent per year, and then display the total spent in 2016 and 207

In [3]:
total_spend_per_year <- tapply(myDF$SPEND, myDF$YEAR, sum, na.rm = TRUE)
print(total_spend_per_year[c("2016", "2017")])

    2016     2017 
19051720 19183059 


So in 2016 we see a total of 19,051,720 and in 2017 19,183,059 spent

## Question 4

Here we are going to load the data and look at the average height per country in the olympics

In [4]:
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/olympics/athlete_events.csv")
tapply(myDF$Height, myDF$NOC, mean, na.rm = TRUE)

AFG      AHO      ALB      ALG      AND      ANG      ANT      ANZ 
170.5926 177.2941 173.0000 174.7029 173.7037 178.2041 175.1217 176.7308 
     ARG      ARM      ARU      ASA      AUS      AUT      AZE      BAH 
176.6047 171.9378 173.2195 175.6667 176.8740 177.3697 172.7174 177.4407 
     BAN      BAR      BDI      BEL      BEN      BER      BHU      BIH 
166.1875 176.5990 172.6316 176.1432 173.6833 177.4058 166.5172 178.6339 
     BIZ      BLR      BOH      BOL      BOT      BRA      BRN      BRU 
175.0548 174.2564 174.0000 171.2957 175.8095 177.6371 173.4245 166.1111 
     BUL      BUR      CAF      CAM      CAN      CAY      CGO      CHA 
172.8309 176.0000 179.2766 166.7193 174.9787 179.5068 174.5854 176.5588 
     CHI      CHN      CIV      CMR      COD      COK      COL      COM 
174.9870 172.6027 176.4151 173.6346 175.3582 171.3243 169.9152 170.8667 
     CPV      CRC      CRO      CRT      CUB      CYP      CZE      DEN 
175.8889 174.3048 183.6132      NaN 174.9082 176.6754 178.1694 179.2010 
     DJI      DMA      DOM      ECU      EGY      ERI      ESA      ESP 
173.4194 174.8889 172.2170 171.1429 178.1045 169.9459 169.5607 174.9217 
     EST      ETH      EUN      FIJ      FIN      FRA      FRG      FSM 
179.3019 170.2301 177.0696 177.4203 175.7579 175.2547 177.6418 161.2400 
     GAB      GAM      GBR      GBS      GDR      GEO      GEQ      GER 
171.5000 170.5500 175.7225 169.5000 175.5531 175.0105 170.2609 177.3445 
     GHA      GRE      GRN      GUA      GUI      GUM      GUY      HAI 
173.8812 176.7833 175.1667 171.1489 172.8871 176.5854 174.4762 173.9306 
     HKG      HON      HUN      INA      IND      IOA      IRI      IRL 
172.2144 173.5468 175.8815 167.4882 171.6386 177.0294 174.8490 176.6143 
     IRQ      ISL      ISR      ISV      ITA      IVB      JAM      JOR 
173.8896 181.2454 173.2453 177.9333 175.2989 178.6316 176.1685 173.1884 
     JPN      KAZ      KEN      KGZ      KIR      KOR      KOS      KSA 
168.2282 175.4657 171.9815 178.0750 166.9091 171.9923 173.5000 173.1444 
     KUW      LAO      LAT      LBA      LBR      LCA      LES      LIB 
174.5000 166.4314 179.6445 168.4655 173.6111 174.8889 167.2549 173.3632 
     LIE      LTU      LUX      MAD      MAL      MAR      MAS      MAW 
175.2826 182.0995 173.8726 168.4643 167.2500 173.5932 170.4903 165.9500 
     MDA      MDV      MEX      MGL      MHL      MKD      MLI      MLT 
176.0180 163.7619 171.9121 169.1335 168.5385 174.8904 180.6066 169.9186 
     MNE      MON      MOZ      MRI      MTN      MYA      NAM      NBO 
183.5638 176.2609 172.6316 174.3423 173.6071 166.1064 172.8592      NaN 
     NCA      NED      NEP      NFL      NGR      NIG      NOR      NRU 
169.6667 179.4889 163.9059 170.0000 174.7183 172.4000 178.0024 167.1818 
     NZL      OMA      PAK      PAN      PAR      PER      PHI      PLE 
177.8329 169.6562 173.9101 171.8824 176.5842 173.5452 170.2681 169.6667 
     PLW      PNG      POL      POR      PRK      PUR      QAT      RHO 
165.4583 170.4257 175.2650 173.5250 161.5580 177.3762 178.3256 169.3333 
     ROT      ROU      RSA      RUS      RWA      SAA      SAM      SCG 
169.9167 172.5536 177.7854 176.2144 167.9167 159.6667 174.8261 187.5114 
     SEN      SEY      SGP      SKN      SLE      SLO      SMR      SOL 
180.1789 175.5000 171.0766 175.2051 177.0396 177.2267 176.5287 169.2308 
     SOM      SRB      SRI      SSD      STP      SUD      SUI      SUR 
176.0000 184.6352 169.6260 173.3333 172.3333 173.9022 175.2216 171.3684 
     SVK      SWE      SWZ      SYR      TAN      TCH      TGA      THA 
178.1702 177.9421 171.5273 174.4545 169.6376 174.9180 179.3056 168.3542 
     TJK      TKM      TLS      TOG      TPE      TTO      TUN      TUR 
174.5692 173.2453 161.4444 172.6981 169.6224 176.9741 178.5284 173.8147 
     TUV      UAE      UAR      UGA      UKR      UNK      URS      URU 
165.7500 172.6538 181.0000 172.0583 174.1875      NaN 175.0632 176.3536 
     USA      UZB      VAN      VEN      VIE      VIN      VNM 

Here we can see the average height of each country that competed in the olympics

## Question 5

Here we take a similar approach as in Q4 but instead we will categorize by sport and when we display we will sort the height

In [5]:
averageHeightPerSport <- tapply(myDF$Height, myDF$Sport, mean, na.rm = TRUE)

# Sort the average results by height
sort(averageHeightPerSport, decreasing = TRUE)

Basketball                Volleyball          Beach Volleyball 
                 190.8699                  186.9948                  186.1450 
               Water Polo                    Rowing                  Handball 
                 184.8346                  184.1722                  183.3846 
                 Baseball                Tug-Of-War                 Bobsleigh 
                 182.5993                  182.4800                  181.4375 
             Motorboating                Ice Hockey                    Tennis 
                 181.0000                  178.9013                  178.8981 
                 Swimming                  Canoeing              Jeu De Paume 
                 178.5625                  178.5396                  178.5000 
                  Sailing         Modern Pentathlon                   Fencing 
                 178.2622                  177.9443                  177.1642 
                Taekwondo                      Luge           Nordic Combined 
                 176.7508                  176.6577                  176.5045 
              Ski Jumping                 Athletics                  Skeleton 
                 176.4028                  176.2563                  176.1886 
                  Cycling                     Rugby                  Racquets 
                 176.1088                  176.0968                  176.0000 
                     Polo                  Football              Rugby Sevens 
                 175.5000                  175.4022                  175.3636 
         Art Competitions             Equestrianism                      Golf 
                 174.6441                  174.3753                  174.2901 
                  Curling                      Judo                 Badminton 
                 174.2031                  174.1874                  174.1788 
            Speed Skating                  Biathlon                  Lacrosse 
                 174.0834                  174.0348                  174.0000 
                Triathlon                  Shooting             Alpine Skiing 
                 173.6458                  173.5722                  173.4891 
                   Hockey      Cross Country Skiing                   Archery 
                 173.3597                  173.2492                  173.2031 
             Snowboarding                    Boxing                 Wrestling 
                 173.0856                  172.8257                  172.3586 
             Table Tennis          Freestyle Skiing Short Track Speed Skating 
                 171.2538                  171.0130                  170.1082 
                 Softball     Synchronized Swimming            Figure Skating 
                 169.3951                  168.4815                  168.2022 
      Rhythmic Gymnastics             Weightlifting                    Diving 
                 167.8703                  167.8248                  166.6343 
             Trampolining                Gymnastics 
                 166.5828                  162.9360

So we have gathered that the tallest sport is Basketball, followed by volleyball. Both of these make sense in terms of what is needed to be competitive in the sports. Likewise it's not surprising that Gy

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.